cds-feature-attachments/src/main/java/com/sap/cds/feature/attachments/handler/applicationservice/helper/ReadonlyDataContextEnhancer.java

```diff
@@ -0,0 +1,46 @@
+package com.sap.cds.feature.attachments.handler.applicationservice.helper;
+
+import java.util.List;
+import java.util.Objects;
+
+import com.sap.cds.CdsData;
+import com.sap.cds.CdsDataProcessor.Validator;
+import com.sap.cds.feature.attachments.generated.cds4j.com.sap.attachments.Attachments;
+import com.sap.cds.feature.attachments.handler.common.ApplicationHandlerHelper;
+import com.sap.cds.services.EventContext;
+
+public final class ReadonlyDataContextEnhancer {
+
+	private static final String CREATE_READONLY_CONTEXT = "CREATE_READONLY_CONTEXT";
+
+	public static void enhanceReadonlyDataInContext(EventContext context, List<CdsData> data) {
+		var filter = ApplicationHandlerHelper.buildFilterForDocumentId();
+		Validator validator = (path, element, value) -> {
+			var documentId = path.target().values().get(Attachments.DOCUMENT_ID);
+			var statusCode = path.target().values().get(Attachments.STATUS_CODE);
+			var scannedAt = path.target().values().get(Attachments.SCANNED_AT);
+			var cdsData = CdsData.create();
+			cdsData.put(Attachments.DOCUMENT_ID, documentId);
+			cdsData.put(Attachments.STATUS_CODE, statusCode);
+			cdsData.put(Attachments.SCANNED_AT, scannedAt);
+			path.target().values().put(CREATE_READONLY_CONTEXT, cdsData);
```

Comment:
If you store everything in a fake data property I think you need to make sure that you clear this `CREATE_READONLY_CONTEXT` property, if the ThreadLocal is not set, to prevent somebody setting it from outside.

Reply:
habe es angepasst und Tests dafür geschrieben